tests/testthat/testImportNetMultinet.r

```diff
@@ -0,0 +1,200 @@
+    
+    # correct log file
+    thisLog <- readLines(logName)
+    expect_true(grepl("event \\| title  \\| importNet.Multinet",   thisLog[1]))
+    expect_true(grepl("event \\| time",                            thisLog[2]))
+    
+    # call
+    expect_true(grepl("^event \\| call   \\| importNet.Multinet\\(", thisLog[3]))
+    expect_true(grepl(paste0("fname = \"", fN, "\""),                thisLog[3]))
+    expect_true(grepl("net = \"PPI\"",                               thisLog[3]))
+    expect_true(grepl("verbose = True",                              thisLog[3]))
+    expect_true(grepl("silent = TRUE",                               thisLog[3]))
+    expect_true(grepl("writeLog = TRUE",                             thisLog[3]))
+    expect_true(grepl(")$",                                          thisLog[3]))
+    
+    expect_true(grepl("gG object has 5 vertices and 4 edges.",     thisLog[6]))
+    expect_true(grepl("\"gG\"",                                    thisLog[7]))
+    expect_true(all(grepl("event \\| output \\| attribute \\|", thisLog[8:11])))
+    expect_true(grepl("class        | \"igraph\"",                 thisLog[8]))
+    expect_true(grepl("type         | \"gG\"",                     thisLog[9]))
+    expect_true(grepl("version      | \"1.0\"",                    thisLog[10]))
+    expect_true(grepl("UUID",                                      thisLog[11]))
+    expect_true(grepl("event | end",                               thisLog[12]))
+    expect_true(grepl("^$",                                        thisLog[13]))
+})
+
+
+test_that("importNet.Multinet() doesn't leak output if silent = TRUE", {
+    fN <- "Multinet.interactions.network_presence.txt"
+    testF <- tempfile()
+    capture.output(gG <- importNet.Multinet(fName = fN,
+                                            net = "PPI",
+                                            verbose = TRUE,
+                                            silent = TRUE,
+                                            writeLog = FALSE),
+                   file = testF)
+    expect_equal(length(readLines(testF)), 0)
+    expect_true(file.remove(testF))
+})
+
+
+test_that("importNet.Multinet() doesn't produce gG from corrupt data", {
+    fN <- "Multinet.interactions.network_presence.txt"
+    
+    # ==== stop on missing NL on last line
+    inputdata <- file(fN, open = "r")
+    expect_error(
+      tail(inputdata, 1) <- !c("\n"),
+      '.checkArgs> stop on missing newline on last line')
+    
+    # ==== stop on NA in requested network
+    network <- list("PPI", "METABOLIC", "GENETIC", "PHOSPHORYLATION",
+                    "REGULATORY", "SIGNALING", "NUM_NETWORKS")
+    expect_error(
+      NotAvailable <- !("net" %in% names(network)),  # "net" input in importNet.Multinet()
+      '.checkArgs> stop on not available in requested network')
+    
+    # ==== stop on mismatch between header elements and data columns
+    inputdata <- file(fN, open = "r")
```

Comment:
I have no idea what you are doing here, and where you got this code from. Please explain.